testgrid/config/config.proto

```diff
@@ -0,0 +1,109 @@
+syntax = "proto3";
+
+package cloud.tools.testgrid.configs.external.kubernetes;
+
+// Protocol buffer for configuring testgrid.k8s.io
+
+// Specifies a group of tests to gather.
+message TestGroup {
+  // Name of this TestGroup, for mapping dashboard tabs to tests.
+  string name = 1;
+
+  // Path to the test result stored in gcs
+  string gcs_prefix = 2;
+
+  // Number of days of test results to gather and serve.
+  int32 days_of_results = 3;
+
+  enum TestsName {
+	  TESTS_NAME_MIN = 0;
+    TESTS_NAME_IGNORE = 1;
+    TESTS_NAME_REPLACE = 2;
+    TESTS_NAME_APPEND = 3;
+  }
+
+  // What to do with the 'Tests name' configuration value. It can replace the
+  // name of the test, be appended to the name of the test, or ignored. If it is
+  // ignored, then the name of the tests will be the build target.
+  TestsName tests_name_policy = 6;
+
+  // Custom column headers for defining extra column-heading rows from values in
+  // the test result.
+  message ColumnHeader {
+      string configuration_value = 3;
+  }
+  repeated ColumnHeader column_header = 9;
+
+  // deprecated - always set to true
+  bool use_kubernetes_client = 24;
+
+  // deprecated - always set to true
+  bool is_external = 25;
+}
+
+
+// Specifies a dashboard.
+message Dashboard {
+  // A list of the tabs on the dashboard.
+  repeated DashboardTab dashboard_tab = 1;
+  // A name for the Dashboard.
+  string name = 2;
+}
+
+message LinkTemplate {
+  // The URL template.
+  string url = 1;
+  // The options templates.
+  repeated LinkOptionsTemplate options = 2;
+}
+
+// A simple key/value pair for link options.
+message LinkOptionsTemplate {
+  // The key for the option. This is not expanded.
+  string key = 1;
+  // The value for the option. This is expanded the same as the LinkTemplate.
+  string value = 2;
+}
+
+// A single tab on a dashboard.
+message DashboardTab {
+  // The name of the dashboard tab to display in the client.
+  string name = 1;
+
+  // The name of the TestGroup specifying the test results for this tab.
+  string test_group_name = 2;
+
+  // Default bug component for manually filing bugs from the dashboard
+  int32 bug_component = 3;
+
+  // Default code search path for changelist search links
+  string code_search_path = 4;
+
+  // The URL template to visit after clicking on a cell.
+  LinkTemplate open_test_template = 7;
+
+  // The URL template to visit when filing a bug.
+  LinkTemplate file_bug_template = 8;
+
+  // The URL template to visit when attaching a bug
+  LinkTemplate attach_bug_template = 9;
+
+  // Text to show in the about menu as a link to another view of the results.
+  string results_text = 10;
+
+  // The URL template to visit after clicking.
+  LinkTemplate results_url_template = 11;
+
+  // The URL template to visit when searching for changelists.
+  LinkTemplate code_search_url_template = 12;
+}
+
+// A service configuration consisting of multiple test groups and dashboards.
+message Configuration {
```

Comment:
What do you think about:

1) add a config field for the default, aka `TestGroup default_test_group = 3`
2) use this value to add missing fields from each entry in test_groups
3) Delete all the `- <<: *DEFAULT_TEST_GROUP` lines
4) do the same for the dashboards

This has two benefits:

1) It reduces the size of config.yaml by roughly a third
2) We no longer ask people to learn complex yaml syntax (anchors) in order to understand the testgrid config.